data-reporting-service/observation-service/src/main/java/gov/cdc/etldatapipeline/observation/service/ObservationService.java

```diff
@@ -63,18 +69,38 @@ private String processObservation(String value) {
                 JsonNode afterNode = payloadNode.get("after");
                 if (afterNode != null && afterNode.has("observation_uid")) {
                     String observationUid = afterNode.get("observation_uid").asText();
+                    observationKey.setObservationUid(Long.valueOf(observationUid));
                     logger.debug(topicDebugLog, observationUid, observationTopic);
                     Optional<Observation> observationData = iObservationRepository.computeObservations(observationUid);
                     if(observationData.isPresent()) {
+                        ObservationReporting reportingModel = modelMapper.map(observationData.get(), ObservationReporting.class);
                         ObservationTransformed observationTransformed = processObservationDataUtil.transformObservationData(observationData.get());
-                        kafkaTemplate.send(observationTopicOutputTransformed, observationTransformed.toString());
+                        buildReportingModelForTransformedData(reportingModel, observationTransformed);
+                        pushKeyValuePairToKafka(observationKey, reportingModel, observationTopicOutputReporting);
+                        return objectMapper.writeValueAsString(observationData.get());
                     }
-                    return objectMapper.writeValueAsString(observationData.get());
                 }
             }
         } catch (Exception e) {
             logger.error("Error processing observation: {}", e.getMessage());
         }
         return null;
     }
+
+    // This same method can be used for elastic search as well and that is why the generic model is present
+    private void pushKeyValuePairToKafka(ObservationKey observationKey, Object model, String topicName) {
```

Comment:
Is there an elastic search model generated for observation

Reply:
not at this moment as we're not using it